src/actions.js

```diff
@@ -179,4 +222,102 @@ export function updateTimeseriesMetadata(uuid) {
   };
 }
 
+export function getTimeseriesMetadataAction(uuid) {
+  return (dispatch, getState) => {
+    const timeseriesState = getState().timeseries;
+
+    if (!timeseriesState[uuid]) {
+      return getTimeseries(uuid).then(results => {
+        if (results && results.length) {
+          dispatch(addTimeseries(uuid, results[0]));
+          return results[0];
+        }
+      });
+    } else {
+      return Promise.resolve(timeseriesState[uuid]);
+    }
+  };
+}
+
+export function getTimeseriesEvents(uuid, start, end, params) {
+  return (dispatch, getState) => {
+    const timeseriesEvents = getState().timeseriesEvents;
+    const events = timeseriesEvents[uuid];
+
+    if (events && events.start === start && events.end === end) {
+      return; // Up to date.
+    } else if (!events || !events.isFetching) {
+      // Fetch it
+      dispatch(fetchTimeseriesEventsAction(uuid, start, end));
+
+      getTimeseries(uuid, start, end, params).then(results => {
+        if (results && results.length === 1) {
+          const result = results[0];
+
+          // Events
+          dispatch(
+            receiveTimeseriesEventsAction(uuid, start, end, result.events)
+          );
+        }
+      });
+    }
+  };
+}
+
+export function getTimeseriesAction(uuid, start, end, params) {
+  return dispatch => {
+    dispatch(getTimeseriesMetadataAction(uuid)).then(timeseriesMetadata =>
+      dispatch(getTimeseriesEvents(timeseriesMetadata.uuid, start, end, params))
+    );
+  };
+}
+
+export function getRasterEvents(raster, geometry, start, end) {
+  return (dispatch, getState) => {
+    if (!raster) return null;
+    const geomKey = `${geometry.coordinates[0]}-${geometry.coordinates[1]}`;
```

Comment:
Can you add a short explanatory comment about this `geomKey` for future reference?